TidepoolKit/Model/Datum/TCalculatorDatum.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  TCalculatorDatum.swift
+//  TidepoolKit
+//
+//  Created by Darin Krauss on 11/1/19.
+//  Copyright © 2019 Tidepool Project. All rights reserved.
+//
+
+import Foundation
+
+public class TCalculatorDatum: TDatum, Decodable {
+    public typealias BloodGlucoseTarget = TBloodGlucose.Target
+    public typealias Units = TBloodGlucose.Units
+
+    public var insulinOnBoard: Double?
+    public var bloodGlucoseInput: Double?
+    public var insulinSensitivity: Double?
+    public var carbohydrateInput: Double?
+    public var insulinCarbohydrateRatio: Double?
+    public var bloodGlucoseTarget: BloodGlucoseTarget?
+    public var recommended: Recommended?
+    public var bolus: TBolusDatum?      // NOTE: API write only
+    public var bolusId: String?         // NOTE: API read only
+    public var units: Units?
```

Comment:
[nit] is units too generic, since there are a number of different measurements/observations included?

Reply:
Name-wise? Yes, I agree. However, this matches the current Tidepool data model. I didn't want to get too much into renaming properties different from the Tidepool data model. (I did expand some abbreviations for some properties in some types. For example, `bgTarget` expanded to `bloodGlucoseTarget`. These expansions are already marked as `TODO`s on the backend platform code so I figured it was reasonable.)